### PR TITLE
Switch deployment to GitHub Actions

### PR DESCRIPTION
**Pull request by parfeon**
## Switch deployment to GitHub Actions

Switch product deployment to GitHub Actions.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          token: ${{ secrets.GH_TOKEN }}
+          path: .github/.release/actions
+      - id: check
+        name: Check pre-release completed
+        uses: ./.github/.release/actions/actions/checks/release
+        with:
+          token: ${{ secrets.GH_TOKEN }}
+  publish:
+    name: Publish package
+    runs-on: ubuntu-latest
```
Comment (crimsonred) [inline .github/workflows/release.yml:32]: UnityEditor is not available for Linux, this should be macOS

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+concurrency: unity_run
+
+on:
+  push:
+  workflow_dispatch:
+
+jobs:
+  tests:
+    name: Run Unity tests
+    runs-on: macos-latest
```
Comment (crimsonred) [inline .github/workflows/run-tests.yml:11]: Need to add support for windows testing as well. I will share the steps with you separately.

**Review by crimsonred — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (parfeon) [reply .github/workflows/release.yml:32]: Yep, missed this.

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (parfeon) [reply .github/workflows/run-tests.yml:11]: @crimsonred so we figured out, that this can't be done on GitHub Action container with Windows Server on it.

Comment (crimsonred) [reply .github/workflows/run-tests.yml:11]: yep :(

**Review by crimsonred — DISMISSED** [review]

**Review by crimsonred — APPROVED** [review]

**Pull request merged**